libdevcore/CMakeLists.txt

```diff
@@ -2,9 +2,7 @@ file(GLOB sources "*.cpp")
 file(GLOB headers "*.h")
 
 add_library(devcore ${sources} ${headers})
-target_link_libraries(devcore PRIVATE ${JSONCPP_LIBRARY} ${Boost_FILESYSTEM_LIBRARIES} ${Boost_REGEX_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-target_include_directories(devcore SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})
+target_link_libraries(devcore PRIVATE jsoncpp ${Boost_FILESYSTEM_LIBRARIES} ${Boost_REGEX_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
@chfast isn't this interfering with `cmake/jsoncpp.cmake`? That defines `${JSONCPP_LIBRARY}` which isn't used anymore.

Reply:
Yeah... Might be good to add it back because I'm not sure what the order will be without it.

Can you try adding
```
target_include_directories(devcore SYSTEM BEFORE PUBLIC ${JSONCPP_INCLUDE_DIR})
```